doc/commands/history.8.rst

```diff
@@ -132,6 +138,59 @@ Examples
 ``dnf5 history undo 4 --ignore-extras``
     | Undo the fourth transaction ignoring extra packages pulled into the reverting transaction.
 
+``dnf5 history list --json``
+    | List all transactions in JSON format for programmatic processing.
+
+``dnf5 history info last --json``
+    | Show detailed info about the last transaction in JSON format with complete package details.
+
+
+JSON Output
+===========
+
+* ``dnf5 history list --json``
+
+The command returns a JSON object with a ``transactions`` array, each describing one transaction.
+Each transaction object contains the following fields:
+
+- ``id`` (integer) - Transaction ID.
+- ``command_line`` (string) - Command line that initiated the transaction.
+- ``start_time`` (string) - Transaction start time in "YYYY-MM-DD HH:MM:SS" format.
+- ``end_time`` (string) - Transaction end time in "YYYY-MM-DD HH:MM:SS" format.
+- ``user_id`` (integer) - User ID who initiated the transaction.
+- ``status`` (string) - Transaction status, typically "Ok".
+- ``releasever`` (string) - System release version when transaction occurred.
+- ``altered_count`` (integer) - Number of packages altered in the transaction.
+
+* ``dnf5 history info --json``
+
+For a single transaction, the command returns a JSON object with a ``transaction`` field.
+For multiple transactions, it returns a JSON object with a ``transactions`` array.
```

Comment:
Unless this logic is hidden in `libdnf5::cli::output::print_transaction_info_json`, I cannot really see it from the `history_info.cpp`

Additionally, I'm not really sure if I like this design… it strikes the same chord as _“empty iterator” vs `null`_, i.e., I, as the consumer of the API, need to differentiate these “edge cases” (that are, in fact, artificial). I would probably prefer to have just `transactions` and don't have this special case for _only one transaction_.